### PR TITLE
Fix models in 2D near the IDL.

### PR DESCRIPTION
**Pull request by bagnell**
Adds another command to each node (when not `scene3DOnly`). Updates that command to position it on the opposite side of the map. Adds the command only if the bounding sphere intersects the IDL.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Can we add a this for this?  Are you writing tests in this branch or `models-2D`?

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -2629,19 +2629,28 @@ define([
...
+                    command2D.boundingVolume = new BoundingSphere();
+                    command2D.modelMatrix = new Matrix4();
+                }
+
                 var nodeCommand = {
                     show : true,
                     boundingSphere : boundingSphere,
                     command : command,
-                    pickCommand : pickCommand
+                    pickCommand : pickCommand,
```
Comment (pjcozzi) [inline Source/Scene/Model.js]: Does picking work for models crossing the IDL?

File: Source/Scene/Model.js [hunk 2]
```diff
@@ -2829,6 +2839,13 @@ define([
                                 Matrix4.clone(command.modelMatrix, pickCommand.modelMatrix);
                                 BoundingSphere.clone(command.boundingVolume, pickCommand.boundingVolume);
                             }
+
+                            command = primitiveCommand.command2D;
+                            if (defined(command) && model._mode === SceneMode.SCENE2D) {
```
Comment (pjcozzi) [inline Source/Scene/Model.js]: Add a comment about why we are doing this.

File: Source/Scene/Model.js [hunk 3]
```diff
@@ -3287,7 +3304,17 @@ define([
                 for (i = 0; i < length; ++i) {
                     nc = nodeCommands[i];
                     if (nc.show) {
-                        commandList.push(nc.command);
+                        var command = nc.command;
+                        commandList.push(command);
+
+                        if (!frameState.scene3DOnly) {
+                            var boundingVolume = command.boundingVolume;
+                            var idl2D = frameState.mapProjection.ellipsoid.maximumRadius * CesiumMath.PI;
```
Comment (pjcozzi) [inline Source/Scene/Model.js]: This is constant per comment, right?  Pull it out of the loop.

**Comment by pjcozzi** [comment]
Just those comments.

**Comment by bagnell** [comment]
>Can we add a this for this? Are you writing tests in this branch or models-2D?

I added tests in `models-2D` so you don't see them in the diff, but I'll add one with a model crossing the IDL here.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look. There are some failing tests that were caused by #3933. I'll fix them in the `models-2D` branch. All of the model tests here should pass though.

File: Source/Scene/Model.js (hunk 1 above)
Comment (pjcozzi) [inline Source/Scene/Model.js]: So picking works across the IDL even though there is only one pick command?

Comment (pjcozzi) [inline Source/Scene/Model.js]: Never mind, I see.

**Comment by pjcozzi** [comment]
Using this code in Sandcastle:
```javascript
var viewer = new Cesium.Viewer('cesiumContainer', {
    infoBox : false,
    selectionIndicator : false
});

function createModel(url, height) {
    viewer.entities.removeAll();

    var position = Cesium.Cartesian3.fromDegrees(-179.0744619, 44.0503706, height);
    var heading = Cesium.Math.toRadians(135);
    var pitch = 0;
    var roll = 0;
    var orientation = Cesium.Transforms.headingPitchRollQuaternion(position, heading, pitch, roll);

    var entity = viewer.entities.add({
        name : url,
        position : position,
        orientation : orientation,
        model : {
            uri : url,
            minimumPixelSize : 128,
            //maximumScale : 20000
        }
    });
    viewer.trackedEntity = entity;
}

var options = [{
    text : 'Aircraft',
    onselect : function() {
        createModel('../../SampleData/models/CesiumAir/Cesium_Air.glb', 5000.0);
    }
}, {
    text : 'Ground vehicle',
    onselect : function() {
        createModel('../../SampleData/models/CesiumGround/Cesium_Ground.glb', 0);
    }
}, {
    text : 'Milk truck',
    onselect : function() {
        createModel('../../SampleData/models/CesiumMilkTruck/CesiumMilkTruck-kmc.glb', 0);
    }
}, {
    text : 'Skinned character',
    onselect : function() {
        createModel('../../SampleData/models/CesiumMan/Cesium_Man.glb', 0);
    }
}];

Sandcastle.addToolbarMenu(options);
```
Switch to 2D, zoom in a bit, then pan to the left.  The model changes orientation and then Cesium throws an exception (only got the exception once though).

ok

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look.

**Comment by pjcozzi** [comment]
JSHint fails in CI:
> Source/Scene/Model.js
  line 3164  col 9  'scratchCartographic' is already defined.

**Comment by pjcozzi** [comment]
Just got this again when panning left to right (back and forth a few times) in 2D using the [above code](https://github.com/AnalyticalGraphicsInc/cesium/pull/3936#issuecomment-220442037).

image

**Comment by bagnell** [comment]
@pjcozzi I can reproduce that error consistently now after talking offline. The problem is in master as well. It happens when the camera lines up with the IDL. I'm working on a fix but it doesn't need to hold up this PR.

**Comment by pjcozzi** [comment]
This is ready since #3953 was merged.

**Pull request merged**